### ISSUE TITLE
Can this be used to something like the autocompletion we have in smartphones' keyboards?

### ISSUE THREAD

**Issue by iacchus** [federico-terzi/espanso#1008]
I would like to make a suggestion of a feature that we could use to emulate the autocompletion and next-possible-word suggestions like as we have in smartphones' keyboards.

Wouldn't it be very cool?

Maybe an espanso's plugin?

**Comment by mrjoecastellon** [comment]
Do you mean like predictive text?

Auto complete of words can be achieved with triggers but you have to set them up yourself. Predictive text would require having language dictionaries and would basically be a whole other scope for this application. 

It's a neat idea though.

**Comment by iacchus** [comment]
Yes, also! I am having some design ideas on it's implementations based on plugins we have in shell and vim.

I don't know if this (like the `h j k l` hinting below) would be straightforward to do in espanso via a plugin with espanso; I am with idea of using menus, like `ibus-typing-booster`. (`ibus-typing-booster`'s proposal is awesome, but it's implementation...)

![idea](https://user-images.githubusercontent.com/881531/156495015-a6b3e130-77cb-4123-b96e-1dfcf94161fe.png)

**Comment by iacchus** [comment]
I've made a little video of `ibus-typing-booster` in action; for those who still don't know, it should be availableon all major GNU/Linux distributions package manager. Idea is great, I'll have in plans checking soon espanso's documentation to see if I can implement something as these ideas... here's it:
 
https://user-images.githubusercontent.com/881531/156495679-ba0a791e-e1b4-49ce-9824-b747ae55204e.mp4

**Comment by federico-terzi** [comment]
Hey @iacchus,

Thank you for the feedback! That sounds definitely interesting, but it would be rather hard to achieve in a cross-platform way. If #915 will be developed, this would become more likely to be possible (as it would use the same APIs as `ibus-typing-booster`).
So stay tuned, someday it might become a reality :)

Cheers :)
